pkg/infrared/infrared.go

```diff
@@ -234,3 +260,39 @@ func (ir *Infrared) copy(dst io.WriteCloser, src io.ReadCloser, srcClosedChan ch
 	io.CopyBuffer(dst, src, *b)
 	srcClosedChan <- struct{}{}
 }
+
+func (ir *Infrared) addConn(c *conn) {
+	ir.mu.Lock()
+	defer ir.mu.Unlock()
+	ir.conns[c.RemoteAddr()] = c
```

Comment:
We don't need mutex lock for this. There won't be any issues with incorrect values if we try to write to this values at the same time. (writing to the same key in this map is the issue, not that it might happen at the same time)

Reply:
I removed the mutex